pkg/securitypolicy/securitypolicyenforcer.go

```diff
@@ -29,16 +31,19 @@ type SecurityPolicyEnforcer interface {
 	EnforceExpectedMountsPolicy(containerID string, spec *oci.Spec) error
 	EnforceMountPolicy(sandboxID, containerID string, spec *oci.Spec) error
 	ExtendDefaultMounts([]oci.Mount) error
+	EnforceLogging() error
```

Comment:
No, it will not grow without bound. It is bounded by the number of things that will be enforced as a constraint. That number will exist no matter what. The number of constraints to enforce will be what it is, whether it is done in the current manner using polymorphic interfaces or via the indirection of the hooks approach where there's still the same number of items but it is now implemented slightly differently. It is a question of what is the best way to maintain. I don't think adding extra indirection is helpful.

Reply:
I was more referring to the lifecycle points that the hooks referenced, to see if the constraints could be divided into groups that run at similar points. 
But if they can be run freely at different points, then that makes sense as well.